application/src/main/java/org/opentripplanner/transit/model/timetable/RealTimeTripTimes.java

```diff
@@ -384,162 +320,20 @@ public int getServiceCode() {
     return scheduledTripTimes.getServiceCode();
   }
 
-  public void setServiceCode(int serviceCode) {
-    this.scheduledTripTimes = scheduledTripTimes
-      .copyOfNoDuplication()
-      .withServiceCode(serviceCode)
-      .build();
+  public RealTimeTripTimes withServiceCode(int serviceCode) {
+    return new RealTimeTripTimes(
+      this,
+      scheduledTripTimes.copyOfNoDuplication().withServiceCode(serviceCode).build()
+    );
```

Comment:
Consider:

```suggestion
  public RealTimeTripTimes withServiceCode(int serviceCode) {
    return new RealTimeTripTimes(
      scheduledTripTimes,
      scheduledTripTimes.copyOfNoDuplication().withServiceCode(serviceCode).build()
    );
```

Reply:
This may not work, because the RealTimeTripTimes may have some properties changed from the schedule. The returned value should only differ in the service code, but nothing else.